app/docs/src/reference/asset/function.md

```diff
@@ -779,77 +780,587 @@ async function main(component: Input): Promise<Output> {
+  domainJson.replace("\n", "\\\\n");
+  const options = {
+    input: `${domainJson}`,
+  };
+  const child = await siExec.waitUntilEnd(
+    "butane",
+    ["--pretty", "--strict"],
+    options,
+  );
+  return {
+    result: child.exitCode === 0 ? "success" : "failure",
+    // NOTE(nick): we probably want both stdout and stderr always, but this will suffice for now.
+    message: child.exitCode === 0 ? child.stdout : child.stderr,
+  };
 }
 ```
 
 Validating that a docker image exists in the registry:
 
 ```typescript
-async function main(
-    component: Input,
-): Promise < Output > {
-    if (!component.domain?.image) {
-        return {
-            result: "failure",
-            message: "no image available",
+async function main(component: Input): Promise<Output> {
+  if (!component.domain?.image) {
+    return {
+      result: "failure",
+      message: "no image available",
+    };
+  }
+  const child = await siExec.waitUntilEnd("skopeo", [
+    "inspect",
+    "--override-os",
+    "linux",
+    "--override-arch",
+    "amd64",
+    `docker://${component.domain.image}`,
+  ]);
+  return {
+    result: child.exitCode === 0 ? "success" : "failure",
+    message:
+      child.exitCode === 0 ? "successly found" : "docker image not found",
+  };
+}
+```
+
+## Management Functions
+
+A management function allows the creation, configuration, and management of
+other components and connections on the graph. They are executed in a change set
+and can enqueue actions to run when the change set is applied to HEAD.
+
+Management functions are typically used to achieve the following things:
+
+1. Importing existing resources into System Initiative
+2. Creation of a template
+3. Configuring pre-existing components
+
+These functions will also be able to be used for a host of other types of use-cases, like:
+
+* Being able to run analysis across a segment of components
+* Being able to manage infrastructure across multiple environments
+
+### Management function arguments
+
+Management functions take an `Input` argument. This argument is an object which
+contains:
+
+- `currentView`
+
+  This is the view in which the management function will execute. This defaults
+  to the `DEFAULT` view on the diagram.
+
+- `thisComponent`
+
+  This is the represention of the component to which the management function is
+  currently running from. In this argument, is the `properties` object, and that
+  will expose `si`, `domain` and `resource` properties as well as the `geometry`
+  of the current component. The geometry is the `height`, `width`, `x` and `y`
+  coordinates for the component.
+
+- `components`
+
+  This is an object that contains all of the components that a management
+  function is connected to, keyed by the component id. Each of these components
+  expose the component type, this is largely the schema name, the `properties`,
+  `geometry`, `parent` and an array `connections`.
+
+The entire structure of the input is:
```

Comment:
Since the type structure automatically adapts if the schemas change, it might be worth pointing out that this type is what it looks like if it's a VPC management component, and that it will change if it manages other kinds of components

Reply:
ah you are correct - I will make it a bit more generic!

Reply:
ok @zacharyhamm made it a bit more generic for now